src/python/pants/net/http/fetcher.py

```diff
@@ -145,19 +145,22 @@ def checksum(self):
       return self._checksum
 
   class ProgressListener(Listener):
-    """A Listener that logs progress to stdout."""
+    """A Listener that logs progress to a stream."""
 
-    def __init__(self, width=None, chunk_size_bytes=None):
+    def __init__(self, width=None, chunk_size_bytes=None, stream=sys.stderr):
```

Comment:
it'd be great to `stream=None` and then `self._stream = stream or sys.stderr` inline in this method.

with the way python default args work, as-is this will latch onto a reference to the value of sys.stderr at the time that this code was imported - which may be the wrong `sys.stderr` after e.g. a daemon/daemon-runner `fork()`. deref'ing the value of `sys.stderr` at runtime vs import time - e.g. in the method body vs as a default arg - will prevent this.